### PR TITLE
refactor(rust): stop deriving `Clone` on `ModuleType`

### PR DESCRIPTION
**Pull request by hyf0**
### Description

preparation for https://github.com/rolldown/rolldown/pull/1632.

### CODE REVIEW

**Comment by hyf0** [comment]
* **#1633** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/1633?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1632** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/1632?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1631** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/1631?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hyf0 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/rolldown_common/src/inner_bundler_options/types/module_type.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@ use serde::Deserialize;
   derive(Deserialize, JsonSchema),
   serde(rename_all = "camelCase", deny_unknown_fields)
 )]
-#[derive(Debug, Clone, Copy)]
```
Comment (IWANABETHATGUY) [inline crates/rolldown_common/src/inner_bundler_options/types/module_type.rs:13]: Can you explain why?

Comment (IWANABETHATGUY) [reply crates/rolldown_common/src/inner_bundler_options/types/module_type.rs:13]: I see the related pr, https://github.com/rolldown/rolldown/pull/1632, such a standalone pull request is confusing.

Comment (hyf0) [reply crates/rolldown_common/src/inner_bundler_options/types/module_type.rs:13]: Yeah. I'll add more descriptions in the future.

**Pull request merged**